Cookbook/Proposals/Base class for FeedbackLoops systems.md

```diff
@@ -0,0 +1,157 @@
+# Base class for FeedbackLoops systems
+
+* Author(s): Sergey Shulga
+* Review Manager: Anders Ha
+
+## Introduction
+
+Almost for all our screens, we implement ViewModels which are powered by the [ReactiveFeeedback](https://github.com/Babylonpartners/ReactiveFeedback).
+
+The current interface of VM looks like this
+
+```swift
+public protocol BoxViewModel: AnyObject {
+    associatedtype State
+    associatedtype Action
+
+    var state: Property<State> { get }
+
+    func send(action: Action)
+}
+```
+
+## Motivation
+
+One of the main issues with this approach
+
+1. We need to implement `input` feedback in pretty much every VM which may lead to errors if do not schedule input correctly, for this we have `Feedback.input()` convenience extension
+
+2. If the screen contains a lot of input field we end up having a lot of event's that are just mutate one property of the `State` e.g
+`Action.didChangeUserName(String)` `Action.didPickBirthDate(Date)`
+ 
+## Proposed solution
+1. Implementation of the feedback:
+
+```swift
+open class FeedbackLoop<State, Event, Action>: FeedbackLoopType {
```

Comment:
can you add example of how the subclass of this will look like and how it will be used in a some view model?

Reply:
```
init(...) {
super.init(
    initial: State(),
    reducer: VM.reduce,
    scheduler: scheduler,
    feedback: [feedbacks]
)
}
```

Reply:
So the view model will be a subclass of `FeedbackLoop`?

Reply:
yes

Reply:
I think these kind of questions should be addressed with extensive code snippets in the document so that we can see the whole picture